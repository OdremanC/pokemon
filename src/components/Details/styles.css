.container {
    width: 40%;
    cursor: default;
    animation: cardSelectedAn 0.8s ease-in-out;
    flex-shrink: 0;
    margin: 0 25% 8%;
    display: flex;
    border: 3px solid yellow;
}

.image {
    width: 50%;
    padding: 20px;
}

.container img {
    width: 100%;
    height: auto;
}
.container h2 {
    text-align: center;
    font-weight: bold;
}

.detail {
    height: 95%;
    margin-left: 20px;
    margin-top: 0;
    padding: 20px 20px 0 20px;
    font-size: 14px;
    border-left: 3px solid yellow;;
    width: 300px;
    animation: detailsAn 1s ease-in-out;
}
.detail h3 {
    font-weight: bold;
}
.detail h4{
    font-weight: bold;
    color: blue;
    cursor: pointer;
    text-align: center;
    margin-top: 20px;
    background-color: yellow;
    padding: 5px 0;
  }

.cardNotSelected {
    opacity: 0;
    width: 0;
    overflow: hidden;
    padding: 0;
    margin: 0;
    height: 0;
    animation: notSelectedAn 0.4s ease-in-out;
}
.cardSelected{
    cursor: default;
    animation: cardSelectedAn 0.8s ease-in-out;
    flex-shrink: 0;
    display: flex;
  }

@keyframes cardSelectedAn {
    0% {width: 14%; flex-shrink: inherit; margin: 0;}
    25% {opacity: 0; width: 23%; flex-shrink: inherit; margin: 0%;}
    50% {width: 30%; flex-shrink: 0; margin: 0 20%;}
    75% {width: 40%; flex-shrink: 0; margin: 0 25%;}
    100% {opacity: 1; flex-shrink: 0; margin: 0 25%}
}
@keyframes detailsAn {
    0% {opacity: 0; height: 0; padding: 0; width: 0;}
    49% {opacity: 0; height: 0; padding: 0; width: 0;}
    50% {opacity: 0; height: auto; padding: 20px; width: 320px;}
    99% {opacity: 0; height: auto; padding: 20px; width: 320px;}
    100% {opacity: 1; height: auto; padding: 20px; width: 320px;}
  }
  @keyframes notSelectedAn {
    0% {
      opacity: 1;
      width: 16%;
      padding: 20px 20px 10px 20px;
      height: auto;
  }
    99% {
      width: 14%;
      padding: 20px 20px 10px 20px;
      height: auto;
    }
    100% {
      opacity: 0;
      width: 0;
      padding: 0;
      height: 0;
    }
  }

  @media (max-width: 812px) and (max-height: 440px) and (orientation: landscape) {
    .card{ width: 19%; margin: 0;}
    .description{padding: 5px;}
  }
  
  @media (max-width: 769px) and (max-height: 1025px) and (orientation: portrait) {
    .container{ 
        width: 100%; 
        margin: 10% 0 20% 0; 
        display: flex; 
        flex-direction: column; 
        align-items: center; 
        justify-content: center;
    }
    .cardSelected { 
        width: 60%; 
        margin-left: 18%;
        margin-right: 0;
        align-items: center;
        display: flex; 
        justify-content: center; 
    }
    .detail {border-left: none; border-top: 3px solid yellow; margin-top: 3%;}
    .detail h3 { font-size: 25px;}
    .detail h4 { font-size: 35px;}
    .image h2 { font-size: 30px;}
    .image { width: 100%;}
  }
  
  @media (max-width: 415px) and (max-height: 738px) and (orientation: portrait) {
    .container{ 
        width: 70%; 
        margin: 0;
        margin-bottom: 7%;
    }
    .description{ padding: 5px; align-items: center;}
    .detail { width: 100%; border-left: none; border-top: 3px solid yellow; margin-top: 3%;}
    .detail h3 { font-size: 20px;}
    .detail h4 { font-size: 25px;}
    .image h2 { font-size: 20px;}
    .image { width: 100%;}
    .cardSelected{
      width: 100%;
      margin: 0;
    }
    .card h2{
      font-size: 14px;
    }
  }