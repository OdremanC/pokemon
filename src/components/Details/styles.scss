
@import '../../assets/styles/mixin';
@import '../../assets/styles/variable.scss';

.pokemon-detail-container {
  position: relative;
}

.pokemon-detail {
  opacity: 0;
  overflow: hidden;
  width: 40%;
  cursor: default;
  margin: 5% auto;
  display: flex;
  transform: translateX(-10%);
  transition: transform .8s ease, opacity .8s ease;
}

.pokemon-detail.cardSelected {
  opacity: 1;
  cursor: default;
  display: flex;
  position: relative;
  animation: showCard 1s ease-in;
  transform: translateX(0);
}

@keyframes showCard {
  0% { opacity: 0;  transform: translateX(-10%);}
  100% { opacity: 1; transform: translateX(0);}
}

@keyframes hideCard {
  0% { opacity: 0;  transform: translateX(-10%);}
  100% { opacity: 1; transform: translateX(0);}
}

.image {
  width: 100%;
}

.pokemon-detail img {
  width: 100%;
  height: auto;
}
.pokemon-detail h2 {
  text-align: center;
  font-weight: bold;
}

.detail {
    height: 95%;
    margin-left: 20px;
    margin-top: 0;
    padding: 0 20px 0 20px;
    font-size: 14px;
    width: 100%;
}
.detail h3 {
    font-weight: bold;
}
.detail h4{
    font-weight: bold;
    color: $blue;
    cursor: pointer;
    text-align: center;
    margin-top: 20px;
    background-color: $yellow;
    padding: 5px 0;
  }

@include media_breakpoint(1024) {
  .pokemon-detail{ 
    width: 50%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 100%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 16px;}
  .image { width: 60%;}
  .cardSelected{
    width: 87%;
    margin: 7% 7%;
  }
  .card h2{
    font-size: 14px;
  }
}
@include media_breakpoint(812) {
  .pokemon-detail{ 
    width: 100%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 100%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 20px;}
  .detail h4 { font-size: 25px;}
  .image h2 { font-size: 20px;}
  .image { width: 25%; margin-left: 10%;}
  .cardSelected{
    width: 100%;
    margin: 10% 0;
  }
  .card h2{
    font-size: 14px;
  }
}
@include media_breakpoint(767) { 
  .pokemon-detail{ 
    width: 80%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 50%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 20px;}
  .detail h4 { font-size: 25px;}
  .image h2 { font-size: 20px;}
  .image { width: 35%; margin-left: 10%;}
  .cardSelected{
    width: 100%;
    margin: 0;
  }
  .card h2{
    font-size: 14px;
  }
}
@include media_breakpoint(608) {
  .pokemon-detail{ 
    width: 50%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 100%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 20px;}
  .detail h4 { font-size: 18px;}
  .image h2 { font-size: 23px;}
  .image { width: 30%;}
  .cardSelected{
    width: 100%;
    margin: 10% 0;
  }
  .card h2{
    font-size: 14px;
  }
}
@include media_breakpoint(401) { 
  .pokemon-detail{ 
    width: 50%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 100%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 14px;}
  .image { width: 30%; margin-top: 10%;}
  .cardSelected{
    width: 100%;
    margin: 10% 0;
  }
}
@include media_breakpoint(479) {
  .pokemon-detail{ 
    width: 50%; 
    margin: 0;      
  }
  .description{ padding: 5px; align-items: center;}
  .detail { width: 100%; border-left: none; margin-top: 3%;}
  .detail h3 { font-size: 14px;}
  .detail h4 { font-size: 12px;}
  .image h2 { font-size: 16px;}
  .image { width: 30%; margin-top: 10%;}
  .cardSelected{
    width: 100%;
    margin: 10% 0;
  }
  .card h2{
    font-size: 14px;
  }
}
