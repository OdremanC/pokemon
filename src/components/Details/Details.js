import React, { useState } from 'react';
import './styles.css';
import notAvailableImage from '../../assets/images/imageNotAvailable.jpg'

function Details({getSelectedPokemon = ()=>{},t=()=>{}, selected={}, detail={}}) {
  const [selectedCard, setSelected] = useState(true);
  const handleSelected = () => {
    getSelectedPokemon();
    setSelected(!selectedCard);
  }

  return (
    <div onClick={(e) => getSelectedPokemon(e)} className={selectedCard ? 'cardSelected' : 'cardNotSelected'}>
      <div className='container'>
        <div className='image'>
          <img src={`https://pokeres.bastionbot.org/images/pokemon/${selected.id}.png`} alt={selected.name} 
            onError={(e)=>{e.target.onerror = null; e.target.src=notAvailableImage}} 
          />
          <h2>{selected.name}</h2>
        </div>
        <div className='detail'>
          <h3>{`${t('Altura')}:`} <span>{`${detail.height / 10}m`}</span></h3>
          <h3>{`${t('Peso')}:`} <span>{`${detail.weight / 10}kg`}</span></h3>
          { 
            detail.abilities && detail.abilities.map(({ability, is_hidden}, key)=> 
            is_hidden ? 
              <h3 key={key}>{`${t('Habilidad normal')}:`} <span>{ability.name}</span></h3> 
              :
              <h3 key={key}>{`${t('Habilidad oculta')}:`} <span>{ability.name}</span></h3>
            )
          }
          <h4 onClick={()=>{ handleSelected() }}>{t('Cerrar')}</h4>
        </div>
      </div>
    </div>
  )
}
export default Details;