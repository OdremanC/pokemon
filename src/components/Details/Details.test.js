import React from 'react';
import ReactDOM from 'react-dom';
import Details from './Details';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<Details />', () => {

    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Details />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshots', () => {
        const wrapper = shallow(<Details/>);
        expect(wrapper).toMatchSnapshot();
    });
})
