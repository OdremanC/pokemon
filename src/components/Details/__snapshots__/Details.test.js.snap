// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> matches snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Details />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pokemon-detail cardSelected"
      >
        <div
          className="image"
        >
          <img
            onError={[Function]}
            src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
          />
        </div>
        <div
          className="detail"
        >
          <h3>
            undefined:
             
            <span>
              NaNm
            </span>
          </h3>
          <h3>
            undefined:
             
            <span>
              NaNkg
            </span>
          </h3>
        </div>
      </div>,
      "className": "pokemon-detail-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image"
          >
            <img
              onError={[Function]}
              src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
            />
          </div>,
          <div
            className="detail"
          >
            <h3>
              undefined:
               
              <span>
                NaNm
              </span>
            </h3>
            <h3>
              undefined:
               
              <span>
                NaNkg
              </span>
            </h3>
          </div>,
        ],
        "className": "pokemon-detail cardSelected",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              onError={[Function]}
              src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
            />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "onError": [Function],
              "src": "https://pokeres.bastionbot.org/images/pokemon/undefined.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                undefined:
                 
                <span>
                  NaNm
                </span>
              </h3>,
              <h3>
                undefined:
                 
                <span>
                  NaNkg
                </span>
              </h3>,
              undefined,
            ],
            "className": "detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "undefined:",
                  " ",
                  <span>
                    NaNm
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "undefined:",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "NaNm",
                  },
                  "ref": null,
                  "rendered": "NaNm",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "undefined:",
                  " ",
                  <span>
                    NaNkg
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "undefined:",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "NaNkg",
                  },
                  "ref": null,
                  "rendered": "NaNkg",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pokemon-detail cardSelected"
        >
          <div
            className="image"
          >
            <img
              onError={[Function]}
              src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
            />
          </div>
          <div
            className="detail"
          >
            <h3>
              undefined:
               
              <span>
                NaNm
              </span>
            </h3>
            <h3>
              undefined:
               
              <span>
                NaNkg
              </span>
            </h3>
          </div>
        </div>,
        "className": "pokemon-detail-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="image"
            >
              <img
                onError={[Function]}
                src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
              />
            </div>,
            <div
              className="detail"
            >
              <h3>
                undefined:
                 
                <span>
                  NaNm
                </span>
              </h3>
              <h3>
                undefined:
                 
                <span>
                  NaNkg
                </span>
              </h3>
            </div>,
          ],
          "className": "pokemon-detail cardSelected",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                onError={[Function]}
                src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
              />,
              "className": "image",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "onError": [Function],
                "src": "https://pokeres.bastionbot.org/images/pokemon/undefined.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  undefined:
                   
                  <span>
                    NaNm
                  </span>
                </h3>,
                <h3>
                  undefined:
                   
                  <span>
                    NaNkg
                  </span>
                </h3>,
                undefined,
              ],
              "className": "detail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "undefined:",
                    " ",
                    <span>
                      NaNm
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "undefined:",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "NaNm",
                    },
                    "ref": null,
                    "rendered": "NaNm",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "undefined:",
                    " ",
                    <span>
                      NaNkg
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "undefined:",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "NaNkg",
                    },
                    "ref": null,
                    "rendered": "NaNkg",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
